broker/lib/jobs/BnRStatusPollerJob.js

```diff
@@ -126,7 +134,104 @@ class BnRStatusPollerJob extends BaseJob {
         Promise.resolve(operationStatusResponse)
       );
   }
+
+  static checkOperationCompletionStatus20(job_data) {
+    const operationName = job_data.operation;
+    const instanceInfo = job_data.operation_details;
+    const instance_guid = instanceInfo.instance_guid;
+    const backup_guid = instanceInfo.backup_guid;
+    const deployment = instanceInfo.deployment;
+    const plan = catalog.getPlan(instanceInfo.plan_id);
+    return Promise.try(() => {
+        if (operationName === 'backup') {
+          return DirectorManager
+            .load(plan)
+            .then(directorManager => directorManager.getServiceFabrikOperationState('backup', instanceInfo));
+        }
+      })
+      .tap(operationStatusResponse => {
```

Comment:
I do not think updating in the tap is a good practice. we should do it under then.

Reply:
done